geth/notification/fcm_provider.go

```diff
@@ -0,0 +1,46 @@
+package notification
+
+import (
+	"github.com/NaySoftware/go-fcm"
+	"github.com/status-im/status-go/geth/notification/message"
+	"github.com/status-im/status-go/geth/common"
+)
+
+// FCMProvider represents messaging provider for notifications.
+type FCMProvider struct {
+	common.FirebaseClient
```

Comment:
I don't understand why we need an interface here. Why not:
```go
type FCMProvider struct {
  client *fcm.Client
}
```

Is it to allow mocking? If so, check if it's possible to customize `fcm.FcmClient`'s server URL. If so, you can use an `httptest.Server`. This interface is not intuitive as it's used only in one place. 

If you decide to leave that interface, let's put it in this package instead of `common`.